src/query/tsdb/remote/server.go

```diff
@@ -113,3 +115,55 @@ func (s *grpcServer) Fetch(
 
 	return err
 }
+
+func (s *grpcServer) Search(
+	message *rpc.SearchRequest,
+	stream rpc.Query_SearchServer,
+) error {
+	var err error
+
+	ctx := retrieveMetadata(stream.Context())
+	logger := logging.WithContext(ctx)
+	searchQuery, err := decodeSearchRequest(message)
+	if err != nil {
+		logger.Error("unable to decode search query", zap.Error(err))
+		return err
+	}
+
+	results, cleanup, err := s.storage.SearchCompressed(
+		ctx,
+		searchQuery,
+		storage.NewFetchOptions(),
+	)
+	defer cleanup()
+	if err != nil {
+		logger.Error("unable to search tags", zap.Error(err))
+		return err
+	}
+
+	pools, err := s.poolWrapper.WaitForIteratorPools(poolTimeout)
```

Comment:
use ctx timeout here ?

Reply:
I'd rather keep this separate; it should never really have to wait since the iterator pools should spin up very soon after startup

Reply:
okay, are you trying to have only on startup ? Then probably using sync.once might be a better way instead of always waiting on a timeout. Also, doesn't look like m3db guys are doing this anywhere. The reason I suggested ctx timeout is so that we have a consistent number per request instead of different functions defining different timeouts.

Reply:
Will change to only on startup. M3 blocks until the pools are intialized, but we want to start m3 asynchronously and take their pools when available.